src/stores/game.js

```diff
@@ -277,12 +284,16 @@ export const useGameStore = defineStore('game', {
     },
     resetPNumIfNull(game) {
       const authStore = useAuthStore();
+      const gameHistoryStore = useGameHistoryStore();
       // Set my pNum if it is null
       if (this.myPNum === null) {
         let myPNum = game.players.findIndex(({ username }) => username === authStore.username);
         if (myPNum === -1) {
           myPNum = null;
         }
+        if (gameHistoryStore.isSpectating) {
+          myPNum = 0;
+        }
```

Comment:
Reinvestigate places we set `myPNum` so that we can consolidate them. I _think_ this should do for most cases and we may not need other assignments